### PR TITLE
Add resource deletion support.

### PR DESCRIPTION
**Pull request by keithburgoyne**

### CODE REVIEW

**Comment by gauthierm** [comment]
Needs a rebase.

File: src/ResourceStore.php [hunk 1]
```diff
@@ -221,6 +221,22 @@ public function save(Resource $resource)
     }
 
     // }}}
+    // {{{ public function delete()
+    
+    public function delete(Resource $resource)
+    {
+        if ($resource->getId() != '') {
```
Comment (charleswaddell) [inline src/ResourceStore.php]: Might make sense to put this into a method on `Resource`. Something like `isNew()` or `isSaved()`.

File: src/ResourceStore.php [hunk 2]
```diff
@@ -221,6 +221,22 @@ public function save(Resource $resource)
...
+            $result = $this->http_client->request(
```
Comment (charleswaddell) [inline src/ResourceStore.php]: This should use `doRequest`.

File: src/ResourceStore.php [hunk 3]
```diff
@@ -221,6 +221,23 @@ public function save(Resource $resource)
...
+    public function delete(Resource $resource)
+    {
+        if ($resource->getId() != '') {
+            $this->doRequest(
+                'DELETE',
+                $this->getResourceAddress(
+                    $resource->getType(),
+                    $resource->getId()
+                ),
+                array()
```
Comment (charleswaddell) [inline src/ResourceStore.php]: We can use the short form of `[]` now.

**Comment by keithburgoyne** [comment]
~~Needs a rebase.~~

Done.

**Comment by keithburgoyne** [comment]
~~Waiting on a fix for the error handling. Currently, it assumes a valid response to a delete request is an error.~~

**Pull request merged**